x-pack/plugins/cases/common/types/api/case/v1.ts

```diff
@@ -104,6 +104,10 @@ export const CasePostRequestRt = rt.intersection([
         limitedStringSchema({ fieldName: 'category', min: 1, max: MAX_CATEGORY_LENGTH }),
         rt.null,
       ]),
+      /**
+       * The list of custom field values of the case.
+       */
+      customFields: CaseCustomFieldsArrayRt,
```

Comment:
Let's restrict the number of total custom fields to five and add some tests for it.